devel/namespaces/index.md

```diff
@@ -0,0 +1,169 @@
+
+* `mozilla-group:<groupName>` -
+   Roles of this form represent the scopes available to members of the given Mozilla LDAP group via the login service.
+
+* `mozilla-user:<userName>` -
+   Roles of this form represent the scopes available to the given Mozilla LDAP user via the login service.
+
+* `repo:<host>/<path>:branch:<branch>`,
+* `repo:<host>/<path>:pull-request` -
+   Roles of this form represent scopes available to version-control pushes and pull requests.
+
+* `moz-tree:level:<level>`
+   Roles of this form include the basic scopes available to version-control trees at each of the three Mozilla source-code managament levels.
+   They are useful as shorthand to configure `repo:*` roles.
+   See [Mozilla Commit Access Policy](https://www.mozilla.org/en-US/about/governance/policies/commit/access-policy/) for information on levels.
+
+* `scheduler-id:<schedulerId>/<taskGroupId>` -
+   Roles of this form represent scopes available to schedulers with the given ID.
+
+* `worker-id:<workerGroup>/<workerId>` -
+   Roles of this form represent scopes available to workers with the given ID.
+
+* `worker-type:<provisionerId>/<workerType>` -
+   Roles of this form represent scopes available to workers of the given type.
+
+## Artifacts
+
+Artifacts are named objects attached to tasks, and available from the queue component.
+Artifact names are, by convention, slash-separated.
+
+* `public/` -
+   The queue allows access to any artifact that begins with `public/` without any kind of authentication.
+   Public names are not further namespaced: tasks can create any public artifacts they like.
+   As such, users should not assume that an artifact with this prefix was created by a known process.
+   In other words, any task can create an artifact named `public/build/firefox.exe` , so do not trust such a file without verifying the trustworthiness of the task.
+
+* `private/` -
+   Artifact names with this prefix are considered non-public, but access to them is otherwise quite broadly allowed (e.g., to all Mozilla employees).
+   In general, users with narrower requirements than "not public" should select a different prefix and add it to this document.
+
+* `index/` -
+   Artifact names with this prefix follow the index namespace, replacing the `.` separators with `/`.
+   For example, a task that indexes artifacts under `gaia.npm_cache.node-v4-2.linux-x64` would name its artifacts with the prefix `index/gaia/npm_cache/node-v4_2/linux-x64`.
+   Scopes for creating and reading such artifacts should follow the ownership hierarchy of the index routes.
+
+* `project/<project>/` -
+   Artifact names with this prefix are the responsibility of the project, which may have further namespace conventions.
+
+## Hooks
+
+Hooks are divided into "hook groups", for which the namespace is defined here.
+Within a hook group, the names are arbitrary (or defined by the project).
+
+* `project:<project>` - hooks for a specific project
```

Comment:
Oh, I missed that this might be broken in hooks... It doesn't really limit identifier format... Which it absolutely should.
We should discuss this... I'll file a bug...